### ISSUE TITLE
Internal error messages and inconsistent type checking behavior with 0.10.2

### ISSUE THREAD

**Issue by nicolabotta** [idris-lang/Idris-dev#3001]
It appears that, with 0.10.2, error messages have further deteriorated. The problem was already there in 0.10.1, see also #2986. But now it seems that I have reached a point at which it has become nearly impossible for me to debug anything but trivial programs. I might be missing something, of course. Consider the following example from https://github.com/nicolabotta/SeqDecProbs/tree/master/frameworks/14- :

```
> import NonNegRational2
> import Fraction
> import NatProperties
> %default total

> zLTs : {m : Nat} -> LT Z (S m)
> zLTs {m} = ltZS m

> x' : Fraction
> x' = (7, fromNat 3 zLTs)
```
The program does not type check:
```
- + Errors (1)
 `-- (no file) line 0 col -1:
     When attempting to perform error reflection, the following internal error occurred:
             No such variable Language.Reflection.PThis is probably a bug. Please consider reporting it at https://github.com/idris-lang/Idris-dev/issues
```
It is not immediately obvious what is going on here but, observing that nothing in the program seems to use non-negative rationals, one could come to the idea of commenting out the `import NonNegRational2` directive on the first line. Attempts at type checking now yields
```
- + Errors (1)
 `-- NonNegRational2Test.lidr line 17 col 3:
     When checking right hand side of x' with expected type
             Fraction
     
     Can't find implementation for Enum (LT 0 (S m) -> Subset Nat Positive)
```
This is much better because now we get some valuable information. Perhaps the culprit is the `fromNat 3 zLTs` computation on the last line of code. Indeed, adding an `import PNatOperations` makes the program type check! So far, so good. Now, we can try to reintroduce the `import NonNegRational2` directive (we finally want to do something with rationals) and extend our program with
```
> x : NonNegRational
> x = classOf x' 
> y : NonNegRational
> y = fromFraction (28, fromNat 8 zLTs)
```
This fails to type check because 
```
- + Errors (1)
 `-- NonNegRational2Test.lidr line 20 col 2:
     When checking right hand side of x with expected type
             NonNegRational
     
     No such variable classOf
```
Now we have a very good error message! `classOf` turns out to be defined in `KernelQuotient`. The file `KernelQuotient.lidr` type checks fine. Unfortunately, when we add `import KernelQuotient` to the above program, we fall back to the initial error message: 
```
- + Errors (1)
 `-- (no file) line 0 col -1:
     When attempting to perform error reflection, the following internal error occurred:
             No such variable Language.Reflection.PThis is probably a bug. Please consider reporting it at https://github.com/idris-lang/Idris-dev/issues
```
At this point, I am lost. Something seems to have happened between 0.9.20 and now that triggers this flood of internal errors. Even when importing modules that, by themselves, type check fine! Especially, with 0.10.2, such internal errors pop-up every second line of code and make it virtually impossible to do any serious work. 

I might have done some mistake porting the framework to 0.10, of course. Any idea of what is happening here? Thanks, Nicola

**Comment by edwinb** [comment]
These are quite possibly to do with some clashes between an old install of 0.9.x and the current install. Have you cleaned all of the ibcs? Possibly also made sure that the old idris is removed from your .cabal directory?

I haven't seen any such internal errors, and it is highly unlikely that this is a new problem with 0.10.

These errors would also be much easier to debug if you gave a sequence of commands that I need to run to reproduce it, assuming I have nothing - starting from the 'git clone'. It may seem trivial, and fairly obvious to work out, but the less time it takes to get from nothing to seeing the error reproduced, the more likely I am to fix it quickly.

**Comment by nicolabotta** [comment]
I apologise Edwin, I was a bit "schlampig" with commits. You should be able to download and unzip the SeqDecProbs repository, then do
```
$ cd SeqDecProbs-master/frameworks/14-/
$ make issue3001
```
This should yield (it takes a while)
```
idris +RTS -K32000000 -RTS -p contrib -p effects --warnreach -V NonNegRational3001.lidr
Type checking ./Basics.lidr
Type checking ./Preorder.lidr
Type checking ./TotalPreorder.lidr
Type checking ./Sigma.lidr
Type checking ./PairsOperations.lidr
Type checking ./NatPredicates.lidr
Type checking ./NatOperations.lidr
Type checking ./NatProperties.lidr
Type checking ./FunOperations.lidr
Type checking ./FunProperties.lidr
Type checking ./IsomorphismOperations.lidr
Type checking ./IsomorphismProperties.lidr
Type checking ./FinOperations.lidr
Type checking ./FinProperties.lidr
Type checking ./TotalPreorderOperations.lidr
Type checking ./Decidable.lidr
Type checking ./VectOperations.lidr
Type checking ./Finite.lidr
Type checking ./FiniteOperations.lidr
Type checking ./SigmaOperations.lidr
Type checking ./ExistsOperations.lidr
Type checking ./Prop.lidr
Type checking ./VectProperties.lidr
Type checking ./FiniteProperties.lidr
Type checking ./SubsetOperations.lidr
Type checking ./Unique.lidr
Type checking ./SubsetProperties.lidr
Type checking ./NatPositive.lidr
Type checking ./PNat.lidr
Type checking ./Fraction.lidr
Type checking ./NumRefinements.lidr
Type checking ./EqualityProperties.lidr
Type checking ./KernelQuotient.lidr
Type checking ./SplitQuotient.lidr
Type checking ./NatDivisor.lidr
Type checking ./NatDivisorOperations.lidr
Type checking ./NatDivisorProperties.lidr
Type checking ./NatGCD.lidr
Type checking ./NatGCDEuclid.lidr
Type checking ./NatGCDOperations.lidr
Type checking ./NatGCDProperties.lidr
Type checking ./NatGCDAlgorithm.lidr
Type checking ./NatCoprime.lidr
Type checking ./NatCoprimeProperties.lidr
Type checking ./PNatOperations.lidr
Type checking ./PNatProperties.lidr
Type checking ./FractionOperations.lidr
Type checking ./FractionNormal.lidr
Type checking ./FractionProperties.lidr
Type checking ./NonNegRational2.lidr
Type checking ./NonNegRational3001.lidr
When attempting to perform error reflection, the following internal error occurred:
        No such variable Language.Reflection.PThis is probably a bug. Please consider reporting it at https://github.com/idris-lang/Idris-dev/issues
Holes: Main.x'
*NonNegRational3001>
```
From this point, you should be able to reproduce the results reported above. Thus, commenting out 
```
> import NonNegRational2
```
and reloading should yield
```
*NonNegRational3001> :r
Type checking ./NonNegRational3001.lidr
NonNegRational3001.lidr:10:4:When checking right hand side of x' with expected type
        Fraction

Can't find implementation for Enum (LT 0 (S m) -> Subset Nat Positive)
Holes: Main.x'
*NonNegRational3001>
```
and so on as reported in my original post. Best, Nicola

**Comment by nicolabotta** [comment]
I have cleaned up my .cabal directory following #2996 and the internal errors is still there with 0.10.2-git:64b44f2

**Comment by edwinb** [comment]
Fixed as in #2986
